websites/C/Coub/presence.ts

```diff
@@ -1,247 +1,370 @@
+                  url: `${document.location.origin}/view/${activeMedia.dataset.permalink}`
+                }
+              ]
+            );
+          if (showBridge)
+            data.buttons.push({
+              label: "Download via Bridge",
+              url: parseBridgeUrl(activeMedia.dataset.permalink)
+            });
+          return data;
+        }
+      },
+      {
+        middleware: (ref) =>
+          !!ref.document.querySelector(".hero-cover[channel-id]"),
+        exec: (
+          context,
+          data,
+          { showWatch, showBridge }: ExecutionArguments
+        ) => {
+          if (!context) return null;
+          const activeMedia = document.querySelector<HTMLElement>(
+            ".coub[coub-block].active"
+          );
+          if (!activeMedia) return null;
+          const title = activeMedia
+              .querySelector(".description__title")
+              .textContent?.trim(),
+            userName = document.querySelector<HTMLHeadingElement>(
+              ".channel__description > h1[title]"
+            ).title,
+            activeTab =
+              document.querySelector<HTMLElement>(
+                ".page__content .page-menu > .page-menu__inner > .page-menu__item.-active"
+              ) ||
+              document.querySelector<HTMLElement>(
+                ".best2020__container .page-menu > .page-menu__inner > .page-menu__item.-active"
+              ) ||
+              document.querySelector<HTMLElement>(
+                ".page__content .story__header"
+              ),
+            isLiked = !!activeMedia.querySelector(
+              ".coub__like-button[widget-like-button].-on"
+            ),
+            activeTabTitle =
+              activeTab?.textContent.trimStart().split("\n")[0]?.trim() ||
+              activeTab?.dataset?.title ||
+              "Feed";
+          if (!title || !userName) return null;
+          data.state = `Browsing ${activeTabTitle} from ${userName}`;
+          data.details = `${title}${isLiked ? " (❤)" : ""}`;
+          const sourceLink = activeMedia.querySelector<HTMLAnchorElement>(
+            ".description__stamp a.description__stamp__source"
+          );
+          data.buttons = [];
+          if (showWatch)
+            data.buttons.push(
+              ...[
+                ...getSourceLink(sourceLink?.href),
+                {
+                  label: "Watch on Coub",
+                  url: `${document.location.origin}/view/${activeMedia.dataset.permalink}`
+                },
+                {
+                  label: "View Profile",
+                  url: `${document.location.origin}/${
+                    document.location.pathname.split("/")[1]
+                  }`
+                }
+              ]
+            );
+          if (showBridge)
+            data.buttons.push({
+              label: "Download via Bridge",
+              url: parseBridgeUrl(activeMedia.dataset.permalink)
+            });
```

Comment:
you can only have 2 buttons, are you sure this will work?

Reply:
> yes, it will override the last button (only for users that want that implementation)